diff --git a/core/modules/dblog/src/Controller/DbLogController.php b/core/modules/dblog/src/Controller/DbLogController.php
index 65874ad8b1d..406aa577097 100644
--- a/core/modules/dblog/src/Controller/DbLogController.php
+++ b/core/modules/dblog/src/Controller/DbLogController.php
@@ -20,6 +20,7 @@
 use Symfony\Component\HttpFoundation\Request;
 use Drupal\Core\Link;
 use Drupal\dblog\DbLogFilters;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
 use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
 
 /**
@@ -49,6 +50,7 @@ class DbLogController extends ControllerBase {
   protected $userStorage;
 
   public function __construct(
+    #[Autowire(service: 'database.dblog')]
     Connection $database,
     DateFormatterInterface $date_formatter,
     FormBuilderInterface $form_builder,
@@ -142,7 +144,6 @@ public function overview(Request $request) {
       'variables',
       'link',
     ]);
-    $query->leftJoin('users_field_data', 'ufd', '[w].[uid] = [ufd].[uid]');
 
     $this->addFilterToQuery($request, $query);
 
@@ -220,8 +221,6 @@ public function eventDetails($event_id) {
     $query = $this->database->select('watchdog', 'w')
       ->fields('w')
       ->condition('w.wid', $event_id);
-    $query->leftJoin('users', 'u', '[u].[uid] = [w].[uid]');
-    $query->addField('u', 'uid', 'uid');
     $dblog = $query->execute()->fetchObject();
 
     if (empty($dblog)) {
diff --git a/core/modules/dblog/src/Hook/DblogHooks.php b/core/modules/dblog/src/Hook/DblogHooks.php
index 5303fcc090e..06d129c0771 100644
--- a/core/modules/dblog/src/Hook/DblogHooks.php
+++ b/core/modules/dblog/src/Hook/DblogHooks.php
@@ -73,7 +73,7 @@ public function cron(): void {
     // Counting the most recent n rows avoids issues with wid number sequences,
     // e.g. auto_increment value > 1 or rows deleted directly from the table.
     if ($row_limit > 0) {
-      $connection = \Drupal::database();
+      $connection = \Drupal\Core\Database\Database::getConnection('default', 'sqlite_logs');
       $min_row = $connection->select('watchdog', 'w')->fields('w', ['wid'])->orderBy('wid', 'DESC')->range($row_limit - 1, 1)->execute()->fetchField();
       // Delete all table entries older than the nth row, if nth row was found.
       if ($min_row) {
