diff --git a/entity_usage.module b/entity_usage.module
index acca35a..61fe3bd 100644
--- a/entity_usage.module
+++ b/entity_usage.module
@@ -110,10 +110,9 @@ function entity_usage_form_alter(&$form, FormStateInterface $form_state, $form_i
     $form['entity_usage_edit_warning'] = [
       '#theme' => 'status_messages',
       '#message_list' => [
-        'warning' => [t('Modifications on this form will affect all <a href="@usage_url" target="_blank">existing usages</a> of this entity.', [
-          '@usage_url' => Url::fromRoute('entity_usage.usage_list', [
-            'entity_type' => $entity->getEntityTypeId(),
-            'entity_id' => $entity->id(),
+        'warning' => [t('Modifications on this form will affect all <a href="@usage_url">existing usages</a> of this entity.', [
+          '@usage_url' => Url::fromRoute('entity.' . $entity->getEntityTypeId() . '.entity_usage', [
+            $entity->getEntityTypeId() => $entity->id(),
           ])->toString(),
         ])],
       ],
@@ -138,10 +137,9 @@ function entity_usage_form_alter(&$form, FormStateInterface $form_state, $form_i
         '#theme' => 'status_messages',
         '#message_list' => [
           'warning' => [
-            t('There are <a href="@usage_url" target="_blank">recorded usages</a> of this entity.', [
-              '@usage_url' => Url::fromRoute('entity_usage.usage_list', [
-                'entity_type' => $entity->getEntityTypeId(),
-                'entity_id' => $entity->id(),
+            t('There are <a href="@usage_url">recorded usages</a> of this entity.', [
+              '@usage_url' => Url::fromRoute('entity.' . $entity->getEntityTypeId() . '.entity_usage', [
+                $entity->getEntityTypeId() => $entity->id(),
               ])->toString(),
             ])
           ],
